app/models/champs/header_section_champ.rb

```diff
@@ -22,6 +22,20 @@ def search_terms
     # The user cannot enter any information here so it doesn’t make much sense to search
   end
 
+  def libelle_with_section_index
+    if libelle_with_section_index? || siblings.any?(&:libelle_with_section_index?)
+      libelle
+    else
+      "#{section_index}. #{libelle}"
+    end
```

Comment:
Pinaillage : j'ai l'impression qu'on peut quasiment exprimer dans le code l'algo que tu expliquais en français dans la PR, en l'écrivant comme ça :

```suggestion
    headers = self + siblings
    if headers.none?(&:libelle_begins_with_number?)
      "#{section_index}. #{libelle}"
    else
      libelle
    end
```

(mais c'est plus pour le style ; dans tous les cas la PR me va bien.)